.github/actions/test-setup/action.yml

```diff
@@ -0,0 +1,19 @@
+name: Testing setup
+descriptions: Checkout the appropriate commit and build it before testing
+
+runs:
+  using: "composite"
+  steps:
+    - if: github.event_name == 'pull_request_target'  # checkout the would-be merge instead of PR branch head
+      uses: actions/checkout@v3
+      with:
+        ref: refs/pull/${{ github.event.number }}/merge
+    # For workflow_dispatch, it's assume that the dispatch branch is already checked out.
+
+    - name: install
+      run: npm ci --cache ./.npm
```

Comment:
Can we please change this to `npm ci --no-fund --no-audit --no-progress`

Reply:
Also, I think we should rename this to `jil-setup/action.yml` and only use it for jobs where we are running jil. We can add another action `ci-setup/action.yml` that can use `npm ci --no-fund --no-audit --no-progress --ignore-scripts`.